File: LeetCode/日常刷题/26-删除有序数组中的重复项.java
class Solution {
    public int removeDuplicates(int[] nums) {
        int n = nums.length;
        int res = 0;
        for (int i = 1; i < n; i++) {
            if (nums[i] == nums[res]) {
                continue;
            } else {
                swap(nums, i, ++res);
            }
        }
        return res+1;
    }

    private void swap(int[] nums, int i, int j) {
        if (i < 0 || i >= nums.length || j < 0 || j >= nums.length) return;
        if (i == j) return;
        nums[i] += nums[j];
        nums[j] = nums[i] - nums[j];
        nums[i] -= nums[j];
        return;
    }
}

File: LeetCode/日常刷题/1-200/26. 删除有序数组中的重复项.py
class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        # 很明显，数组有序，且要求原地很多时候就是需要使用双指针
        if not nums:
            return 0

        l = 0
        n = len(nums)
        for r in range(1, n):
            if nums[r] == nums[l]:
                continue
            l += 1
            nums[l], nums[r] = nums[r], nums[l]
        return l+1